src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1238,6 +1238,22 @@ AddToAttributeEquivalenceClass(AttributeEquivalenceClass *attributeEquivalenceCl
 		return;
 	}
 
+#if PG_VERSION_NUM >= PG_VERSION_16
+
```

Comment:
I actually like the function you added, it was documenting the purpose nicely. Can you add the function again and call it here only? Then, we can still move the ifdefs into the function.